### DISCUSSION TITLE
Enable `Serial2` on the 2 series

### DISCUSSION THREAD

**Discussion by clarkbremer** [SpenceKonde/megaTinyCore#703] [Library discussion]
Using ATtiny824, and naively attempted to set up `Serial2` the same way I would on an old "mega" board.  What's the trick for using the second uart on the "2" series?

**Comment by SpenceKonde** [comment]
There are no tinyavr parts with a Serial2. You want Serial1. There's Serial
and Serial1. Take care if using the .swap() method. There are 2 serial
ports each pf which can be on 1 of 2 pairs of pins, but serial default
pinscand serial1 alt pins are the same pins

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Comment by clarkbremer** [comment]
And of course that works perfectly.  Thanks, Spence.  I was thinking Serial == Serial1.  The ESP32 has 3 uarts, which is why that worked for me in the past.

**Comment by SpenceKonde** [comment]
Yup. On dxcore but I don't think here, Serial is just a #define and the
actual object is called Serial0, so that one could create a variant file
wher default port wasn't the zeroth one. With only two ports sharing 3
pairs of pins one of which is only present on larger parts it seemed silly.

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***
